src/utils/index.ts

```diff
@@ -28,3 +35,28 @@ export function getCellClassname<R, SR>(column: CalculatedColumn<R, SR>, ...extr
     ...extraClasses
   );
 }
+
+export function getColSpan<R, SR>(column: CalculatedColumn<R, SR>, viewportColumns: readonly CalculatedColumn<R, SR>[], args: ColSpanArgs<R, SR>) {
+  const colSpan = typeof column.colSpan === 'function' ? column.colSpan(args) : column.colSpan;
+  if (colSpan && !areColSpanColumnsCompatible(column, viewportColumns, colSpan)) {
+    // ignore colSpan if it spans over frozen and regular columns
+    return undefined;
+  }
+  return colSpan;
+}
+
+function areColSpanColumnsCompatible<R, SR>(column: CalculatedColumn<R, SR>, viewportColumns: readonly CalculatedColumn<R, SR>[], colSpan: number) {
+  const isFrozen = column.frozen;
+  if (!isFrozen) return true;
+  const startIdx = column.idx - viewportColumns[0].idx;
+  for (
```

Comment:
Don't we only need to check for starting column + `columns[starting column index + colspan]`?
Why check for the columns in between?
Shouldn't we also check for out of bounds spanning/negative values/non-integer values?

Reply:
We are only checking columns within the colspan range. 

```
index < colSpan && startIdx + index < viewportColumns.length;
```

Had to add `viewportColumns` as an extra check as it was failing with undefined column. I can check again. 

> Shouldn't we also check for out of bounds spanning/negative values/non-integer values?
Added some validations